libchisel/src/lib.rs

```diff
@@ -27,8 +27,8 @@ pub trait ModuleCreator {
 }
 
 pub trait ModuleTranslator {
-    /// Translates module. Returns new module. Can fail with ModuleError::NotSupported.
-    fn translate(&self, module: &Module) -> Result<Module, ModuleError>;
+    /// Translates module. Returns new module or none if nothing was modified. Can fail with ModuleError::NotSupported.
+    fn translate(&self, module: &Module) -> Result<Option<Module>, ModuleError>;
```

Comment:
In the future I would like to find a better alternative to having the option nested in the result. De-structuring multiple layers becomes tedious.

Reply:
There's plenty of example of this. Not sure what is better, but it may premature optimisation.